library/src/main/java/org/xmtp/android/library/Dm.kt

```diff
@@ -0,0 +1,267 @@
+package org.xmtp.android.library
+
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import org.xmtp.android.library.codecs.ContentCodec
+import org.xmtp.android.library.codecs.EncodedContent
+import org.xmtp.android.library.codecs.compress
+import org.xmtp.android.library.libxmtp.Member
+import org.xmtp.android.library.libxmtp.MessageV3
+import org.xmtp.android.library.messages.DecryptedMessage
+import org.xmtp.android.library.messages.MessageDeliveryStatus
+import org.xmtp.android.library.messages.PagingInfoSortDirection
+import org.xmtp.android.library.messages.Topic
+import org.xmtp.proto.message.api.v1.MessageApiOuterClass
+import uniffi.xmtpv3.FfiConversation
+import uniffi.xmtpv3.FfiConversationMetadata
+import uniffi.xmtpv3.FfiDeliveryStatus
+import uniffi.xmtpv3.FfiListMessagesOptions
+import uniffi.xmtpv3.FfiMessage
+import uniffi.xmtpv3.FfiMessageCallback
+import java.util.Date
+import kotlin.time.Duration.Companion.nanoseconds
+import kotlin.time.DurationUnit
+
+class Dm(val client: Client, private val libXMTPGroup: FfiConversation) {
+    val id: String
+        get() = libXMTPGroup.id().toHex()
+
+    val topic: String
+        get() = Topic.groupMessage(id).description
+
+    val createdAt: Date
+        get() = Date(libXMTPGroup.createdAtNs() / 1_000_000)
+
+    private val metadata: FfiConversationMetadata
+        get() = libXMTPGroup.groupMetadata()
+
+    val name: String
+        get() = libXMTPGroup.groupName()
+
+    val imageUrlSquare: String
+        get() = libXMTPGroup.groupImageUrlSquare()
+
+    val description: String
+        get() = libXMTPGroup.groupDescription()
+
+    val pinnedFrameUrl: String
+        get() = libXMTPGroup.groupPinnedFrameUrl()
```

Comment:
Should we expose some of these things in the dm?

Reply:
It might be easiest for now if these things were not modifiable in DM's, both because they're not available in v2, and because they're harder to carry over between duplicate v3 DM's